feature/experimental/gribi/ate_tests/route_removal_during_failover_test/route_removal_during_failover_test.go

```diff
@@ -0,0 +1,565 @@
+
+// TestRouteRemovalDuringFailover is to test gRIBI flush and slave switchover
+// concurrently, validate reinject of gRIBI programmed routes and traffic.
+func TestRouteRemovalDuringFailover(t *testing.T) {
+	d := &oc.Root{}
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+
+	ctx := context.Background()
+	gribic := dut.RawAPIs().GRIBI().Default(t)
+	dp1 := dut.Port(t, "port1")
+	ap1 := ate.Port(t, "port1")
+	top := ate.Topology().New()
+	// configure DUT port#1 - source port
+	configureSubinterfaceDUT(t, d, dp1, 0, 0, dutPort1.IPv4)
+	configureInterfaceDUT(t, dp1, d, "src")
+	configureATE(t, top, ap1, "src", 0, dutPort1.IPv4, atePort1.IPv4CIDR())
+	pushConfig(t, dut, dp1, d)
+	dp2 := dut.Port(t, "port2")
+	ap2 := ate.Port(t, "port2")
+	// Configure 64 subinterfaces on DUT-ATE- PORT#2
+	subIntfIPs := generateSubIntfPair(t, dut, dp2, ate, ap2, top, d)
+	top.Push(t).StartProtocols(t)
+
+	// Connect gRIBI client to DUT referred to as gRIBI - using PRESERVE persistence and
+	// SINGLE_PRIMARY mode, with FIB ACK requested. Specify gRIBI as the leader.
+	client := fluent.NewClient()
+	client.Connection().WithStub(gribic).WithPersistence().WithInitialElectionID(1, 0).
+		WithFIBACK().WithRedundancyMode(fluent.ElectedPrimaryClient)
+	client.Start(ctx, t)
+	defer client.Stop(t)
+
+	defer func() {
+		// Flush all entries after test.
+		if err := gribi.FlushAll(client); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	client.StartSending(ctx, t)
+	if err := awaitTimeout(ctx, client, t, time.Minute); err != nil {
+		t.Fatalf("Await got error during session negotiation for clientA: %v", err)
+	}
+	eID := gribi.BecomeLeader(t, client)
+
+	t.Log("Flush existing gRIBI routes before test.")
+	if err := gribi.FlushAll(client); err != nil {
+		t.Fatal(err)
+	}
+
+	args := &testArgs{
+		ctx:        ctx,
+		client:     client,
+		dut:        dut,
+		ate:        ate,
+		top:        top,
+		electionID: eID,
+	}
+	// virtualVIPs are ipv4 entries in default vrf
+	pushDefaultEntries(t, args, subIntfIPs)
+
+	// Send traffic from ATE port-1 to prefixes in IPBlock1 and ensure traffic
+	// flows 100% and reaches ATE port-2.
+	flow := createTrafficFlow(t, args.ate, args.top)
+	testTraffic(t, *args, flow)
+
+	controllers := cmp.FindComponentsByType(t, dut, controlcardType)
+	t.Logf("Found controller list: %v", controllers)
+	// Only perform the switchover for the chassis with dual controllers.
+	if len(controllers) != 2 {
+		t.Skipf("Dual controllers required on %v: got %v, want 2", dut.Model(), len(controllers))
+	}
+
+	secondaryBeforeSwitch, primaryBeforeSwitch := findSecondaryController(t, dut, controllers)
+
+	if ok := switchoverReady(t, dut, primaryBeforeSwitch); !ok {
+		t.Fatalf("Controller %q did not become switchover-ready before test.", primaryBeforeSwitch)
+	}
+
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	switchoverRequest := &spb.SwitchControlProcessorRequest{
+		ControlProcessor: &tpb.Path{
+			Elem: []*tpb.PathElem{{Name: secondaryBeforeSwitch}},
```

Comment:
please change this line similar to the existing tests:
```
ControlProcessor: cmp.GetSubcomponentPath(secondaryBeforeSwitch),

Reply:
updated , thank you